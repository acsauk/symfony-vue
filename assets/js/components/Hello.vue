<template>
    <div>
        <div class="section">
            <div class="shopping-list container">
                <h2>Hello {{ name }} from {{ greeter }}!</h2>
                <button v-if="state === 'default'" class="btn btn-primary" @click="changeState('edit')">Add Item</button>
                <button v-else class="btn btn-cancel" @click="changeState('default')">Cancel</button>
            </div>
        </div>

        <div v-if="state === 'edit'" class="add-item-form">
            <input type="text" v-model="newItem" placeholder="Add an item" @keyup.enter="saveItem">
            <p>{{ characterCount }}/200</p>
            <button class="btn btn-primary" :disabled="newItem.length === 0" @click="saveItem">Save Item</button>
        </div>

        <ul>
            <li v-for="item in reversedItems" :class="{strikeout: item.purchased}" @click="togglePurchased(item)">{{ item.label }}</li>
        </ul>
        <p v-if="items.length === 0">No items to buy :(</p>
    </div>
</template>

<script>
    export default {
        data() {
            return {
                state: 'default',
                name: "Vue",
                greeter: "Alex",
                items: [
                    {
                        label: 'Item 1',
                        purchased: true,
                    },
                    {
                        label: 'Item 2',
                        purchased: false,
                    },
                    {
                        label: 'Item 3',
                        purchased: false,
                    }
                ],
                newItem: ''
            }
        },

        computed: {
            characterCount() {
                return this.newItem.length;
            },
            reversedItems() {
                return this.items.slice(0).reverse();
            }
        },

        methods: {
            saveItem: function() {
                this.items.push(
                    {
                        label: this.newItem,
                        purchased: false,
                    }
                );
                this.newItem = '';
            },
            changeState: function(newState) {
                this.state = newState;

                this.newItem = '';
            },
            togglePurchased: function (item) {
                item.purchased = !item.purchased;
            }
        }
    }
</script>

<style scoped>

</style>