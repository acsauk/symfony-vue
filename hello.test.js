import { mount } from "@vue/test-utils";
import Hello from "./assets/js/components/Charges.vue";

describe("Hello component", () => {

    it("renders correctly", () => {
        const comp = mount(Hello);
        expect(comp.html()).toMatchSnapshot();
    });

    it("renders correctly after changing the name state", () => {
        const comp = mount(Hello);
        comp.setData({ name: "Camel" });

        expect(comp.html()).toMatchSnapshot();
    });

    it("renders correctly setting the `greeter` prop", () => {
        const comp = mount(Hello);
        comp.setData({ greeter: 'Joe' });

        expect(comp.html()).toMatchSnapshot();
        expect(comp.data().greeter).toBe("Joe");
    });
});

