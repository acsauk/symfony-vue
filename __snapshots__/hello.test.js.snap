// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component renders correctly 1`] = `"<div>Hello Vue from Alex!</div>"`;

exports[`Hello component renders correctly after changing the name state 1`] = `"<div>Hello Camel from Alex!</div>"`;

exports[`Hello component renders correctly setting the \`greeter\` prop 1`] = `"<div>Hello Vue from Joe!</div>"`;
